ni_measurementlink_service/discovery/_client.py

```diff
@@ -111,6 +111,8 @@ def register_service(self, service_info: ServiceInfo, service_location: ServiceL
         Returns:
             ID that can be used to unregister the service.
         """
+        service_info.annotations["ni/service.programminglanguage"] = "Python"
```

Comment:
`service_info` belongs to the caller. It is not yours to modify.

You should copy the annotations dict, modify the copy, and pass the copy to `discovery_service_pb2.ServiceDescriptor` below.

Note that using the union operator like `annotations=service_info.annotations | {"ni/service.programminglanguage": "Python"}` would require Python 3.9, but this module still supports Python 3.8.